ft-sdk/src/in_.rs

```diff
@@ -16,4 +16,33 @@ impl In {
             form_errors: std::collections::HashMap::new(),
         })
     }
+
+    pub fn user_agent(&self) -> String {
+        self.req
+            .headers()
+            .get(http::header::USER_AGENT)
+            .and_then(|v| v.to_str().map(|v| v.to_string()).ok())
+            .unwrap_or("anonymous".to_string())
+    }
+
+    pub fn json_body<T: serde::de::DeserializeOwned>(&self) -> serde_json::Result<T> {
+        serde_json::from_slice(&self.req.body())
+    }
+
+    pub fn filtered_json_body<T: serde::de::DeserializeOwned>(&self) -> serde_json::Result<T> {
+        // Removing password fields from body (for logging)
```

Comment:
Nopes. You are supposed to log each field individually, and erase things there. Delete this method.